Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -1365,7 +1365,9 @@ gbuild verbosity numJobs pkg_descr lbi bm clbi = do
                                              [tmpDir </> x | x <- cObjs ++ cxxObjs]
                     }
       dynLinkerOpts = mempty {
-                      ghcOptRPaths         = rpaths
+                      ghcOptRPaths         = rpaths,
+                      ghcOptInputFiles     = toNubListR
+                                             [tmpDir </> x | x <- cObjs ++ cxxObjs]
```

Comment:
echoing @phadej's question:

> This change is something else? We don't need cmmObjs nor asmObjs however?

Reply:
I don't really understand the question. Before the files weren't even passed the the linker, now they are. It is possible that these other files are also needed but this patch fixes the issue for cxxObjs, as claimed, and other files types are out-of-scope.

Reply:
> I don't really understand the question.

Rephrasing as I understand it: Is this part of the "Include cmm-sources and asm-sources when linking shared objects" goal? If yes, why aren't cmmObjs and asmObjs included here too?

> this patch fixes the issue for cxxObjs

wasn't it cmm and asm?

Reply:
`cObjs` contains all the objs you list as it is constructed from cLikeFiles (which contains asm, cmm, c etc).

https://github.com/haskell/cabal/blob/5f691229abd511884ce2cca5b33736222b143000/Cabal/src/Distribution/Simple/GHC.hs#L554